ACPs/194-streaming-asynchronous-execution/README.md

```diff
@@ -0,0 +1,394 @@
+2. Gas excess $x$ (and therefore gas price) at the time of execution.
+
+For a queue of transactions $Q = \\{i\\}_ {i \ge 0}$ the gas excess $x_j$ immediately prior to execution of transaction $j \in Q$ is a monotonic, non-decreasing function of the gas usage of all preceding transactions in the queue; i.e. $x_j~:=~f(\\{g_i\\}_{i<j})$.
+
+To see this, consider transaction $0 \le k<j$ consuming gas $g_k$.
+A decrease in $g_k$ reduces the immediate increase of $x$.
+Furthermore, this lowered consumption can never reduce a chunk's gas surplus and hence any decrease of $x$ is $\ge$ predicted.
+The excess, and hence gas price, for every later transaction $x_{i>k}$ is therefore reduced:
+
+$$
+\downarrow g_k \implies
+\begin{cases}
+    \downarrow \frac{g \cdot (p-1)}{p} \\
+    \uparrow \frac{s}{p}
+\end{cases}
+\implies \downarrow \Delta x_k
+\implies \downarrow M \cdot \exp\left(\frac{x_{i>k}}{K}\right)
+$$
+
+Given maximal gas consumption under (1), the monotonicity of $f$ implies (2).
+
+Since we are working with non-negative integers, it follows that multiplying a transaction's gas limit by the hypothetical gas price of (2) results in its worst-case gas cost.
+Any sender able to pay for this upper bound (in addition to value transfers) is guaranteed to be able to pay for the actual execution cost.
+Transaction _acceptance_ under worst-case cost validity is therefore a guarantee of _settlement_.
+
+### Queue DoS protection
+
+Worst-case cost validity only protects against DoS at the point of execution but leaves the queue vulnerable to high-limit, low-usage transactions.
+For example, a malicious user could send a transfer-only transaction (21k gas) with a limit set to multiple chunks.
+Although they would have to have sufficient funds to theoretically pay for all the chunks, they would never actually be charged this amount. Pushing a sufficient number of such transactions to the queue would artificially inflate the worst-case cost of other users.
+
+Gas limits are typically set higher than expected gas used to allow for a buffer should gas estimates be imprecise.
+A lower bound of 50% of the specified limit, for example, would allow for a 100% buffer on estimates without penalising the sender, while still disincentivising falsely high limits.
+
+#### Upper bound on queue DoS
+
+With $R$ (for rate) as already defined (gas capacity per second), we introduce $g_L$, $g_U$, and $g_C$ as gas _limit_, _used_ and _charged_ per transaction, respectively, and
+
+$$
+g_C := \max(g_U, \lambda \cdot g_L)
+$$
+
+where $\lambda$ enforces a lower bound on the charge.
+
+The actual gas excess $x_A$ has an upper bound of the worst-case excess $x_W$, both of which can be used  to calculate respective base fees $f_A$ and $f_W$ (the variable element of gas prices) from the existing exponential function:
+
+$$
+f := M \cdot \exp\left( \frac{x}{K} \right).
+$$
+
+Mallory is attempting to maximize the DoS ratio
```

Comment:
who?

Reply:
[Alice and Bob's arch nemesis](https://en.wikipedia.org/wiki/Alice_and_Bob#:~:text=A%20malicious%20attacker.)!